### PR TITLE
ERR: Better error message for missing columns in aggregate

### LINKED ISSUE #32755: Request: Better error message for missing columns in aggregate
When attempting to aggregate a non-existing column, this error is raised, which is not very descriptive: `SpecificationError: nested renamer is not supported`
```
In [39]: pd.DataFrame([[1, 2]], columns=['A', 'B']).groupby('A').agg({'B': 'mean'})
Out[39]: 
   B
A   
1  2

In [40]: pd.DataFrame([[1, 2]], columns=['A', 'B']).groupby('A').agg({'B': 'mean', 'non-existing': 'mean'})
Out[40]:
...
SpecificationError: nested renamer is not supported
```

### PR DESCRIPTION
**Pull request by JDTruj2018**
More descriptive SpecificationError message that reports to user non-existing columns causing error

- [x] closes #32755
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/base.py [hunk 1]
```diff
@@ -356,7 +356,11 @@ def _aggregate(self, arg, *args, **kwargs):
                 if isinstance(obj, ABCDataFrame) and len(
                     obj.columns.intersection(keys)
                 ) != len(keys):
-                    raise SpecificationError("nested renamer is not supported")
+                    cols = sorted(set(keys) - set(obj.columns.intersection(keys)))
```
Comment (jreback) [inline pandas/core/base.py]: its not worth it dispatching on both, just make. single message

File: doc/source/whatsnew/v1.1.0.rst [hunk 2]
```diff
@@ -84,7 +84,7 @@ Other API changes
 - Added :meth:`DataFrame.value_counts` (:issue:`5377`)
 - :meth:`Groupby.groups` now returns an abbreviated representation when called on large dataframes (:issue:`1135`)
 - ``loc`` lookups with an object-dtype :class:`Index` and an integer key will now raise ``KeyError`` instead of ``TypeError`` when key is missing (:issue:`31905`)
--
+- :meth:`DataFrame.agg` now provides more descriptive SpecificationError message when attempting to aggregating non-existant column (:issue:`32755`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst]: can you move this to reshaping

**Review by jreback — CHANGES_REQUESTED** [review]
looks fine, ping on green.

**Comment by jreback** [comment]
merge master to fix the docs issue

File: pandas/core/base.py (hunk 1 above)
Comment (JDTruj2018) [reply pandas/core/base.py]: Fixed both of your suggestions - am getting a lot of seemingly unrelated failures on pandas/tests/groupby/aggregate/test_other.py though. Should I push anyway?

**Comment by JDTruj2018** [comment]
@jreback Made changes and green

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @JDTruj2018

**Pull request merged**